nap/utils.py

```diff
@@ -1,9 +1,24 @@
 from __future__ import unicode_literals
 import itertools
+from urllib import urlencode
+from urlparse import parse_qsl, urlsplit, urlunsplit
+
 import six
 from six.moves import urllib
 
 
+def normalize_url(url_string):
+    url = urlsplit(url_string)
+    if url.query:
+        query_params = sorted(parse_qsl(url.query, keep_blank_values=True), key=lambda key_value: key_value[0])
+        query_normalized = urlencode(query_params)
+        url = url._replace(query=query_normalized)
```

Comment:
This is actually not a private method. Python uses underscore for `namedtuple` methods so that they do not conflict with attribute names that you want to use. Despite starting with an underscore, they are written with full intentions for us, the `namedtuple` end-user, to use them.

See [namedtuple](https://docs.python.org/2/library/collections.html#namedtuple-factory-function-for-tuples-with-named-fields):

> To prevent conflicts with field names, the method and attribute names start with an underscore.